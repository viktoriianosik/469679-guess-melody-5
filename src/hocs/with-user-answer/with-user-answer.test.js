import React from "react";
import PropTypes from "prop-types";
import renderer from "react-test-renderer";
import withUserAnswer from "./with-user-answer";

const noop = () => {};

const question = {
  type: `genre`,
  genre: `rock`,
  answers: [{
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `rock`,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `blues`,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `jazz`,
  }, {
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    genre: `rock`,
  }],
};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withUserAnswer(MockComponent);

it(`withUserAnswer render correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      question={question}
      onAnswer={noop}
    >
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
