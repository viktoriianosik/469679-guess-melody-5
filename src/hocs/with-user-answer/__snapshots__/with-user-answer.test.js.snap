// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withUserAnswer render correctly 1`] = `<div />`;
